### PR TITLE
Integrations Bundle: Allow developers process created/updated pulled objects

### PR DESCRIPTION
**Pull request by kuzmany**
| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" for all features, enhancements and bug fixes (until 3.3.0 is released) 
| Bug fix?                               | yes/no
| New feature?                           | yes/no
| Deprecations?                          | yes/no
| BC breaks?                             | yes/no
| Automated tests included?              | yes/no
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
This PR allow work with created/update objects after pull from integration. For example we need make API call after that with contact ID and integration ID.
This is reason why we add also integration ID to UpdatedObjectMappingDAO https://github.com/mautic/mautic/pull/9687/files#diff-d59a37a0cfdcf2cf90decf5ca107c5b45fb6d7dfc02f96aa3a212c5d3510b24eR52

This PR also add internalObjectId to UpdatedObjectMappingDAO - we need work after update process some API related to internalId and integrationId https://github.com/mautic/mautic/pull/9687/files#diff-d59a37a0cfdcf2cf90decf5ca107c5b45fb6d7dfc02f96aa3a212c5d3510b24eR52

Example fo subscriber with lower level and process already created/update objects:

```PHP
class ContactObjectSubscriber implements EventSubscriberInterface
{

    /**
     * @return array
     */
    public static function getSubscribedEvents()
    {
        return [
            IntegrationEvents::INTEGRATION_UPDATE_INTERNAL_OBJECTS => ['updateContacts', -1],
            IntegrationEvents::INTEGRATION_CREATE_INTERNAL_OBJECTS => ['createContacts', -1],
        ];
    }

    public function updateContacts(InternalObjectUpdateEvent $event): void
    {
        if (Contact::NAME !== $event->getObject()->getName()) {
            return;
        }

        // do whatever you want with $event->getUpdatedObjectMappings()
    }

    public function createContacts(InternalObjectCreateEvent $event): void
    {
        if (Contact::NAME !== $event->getObject()->getName()) {
            return;
        }
        // work whatever you want with $event->getObjectMappings()
    }
}
```

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. This need integration based on integration framework
3. Code review would be good

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/9687"></a>

### CODE REVIEW

File: app/bundles/IntegrationsBundle/EventListener/CompanyObjectSubscriber.php
Comment (dennisameling) [inline app/bundles/IntegrationsBundle/EventListener/CompanyObjectSubscriber.php:75]: @kuzmany could you explain why we need to remove this line? I couldn't find it in the PR description. Thanks!

Comment (kuzmany) [reply app/bundles/IntegrationsBundle/EventListener/CompanyObjectSubscriber.php:75]: I describe it in description. We made integration what need process objects after processed it by IntegrationsBundle. 
We need created/updated object Ids post to another API. 

```php
class ContactObjectSubscriber implements EventSubscriberInterface
{

    /**
     * @return array
     */
    public static function getSubscribedEvents()
    {
        return [
            IntegrationEvents::INTEGRATION_UPDATE_INTERNAL_OBJECTS => ['updateContacts', -1],
            IntegrationEvents::INTEGRATION_CREATE_INTERNAL_OBJECTS => ['createContacts', -1],
        ];
    }

    public function updateContacts(InternalObjectUpdateEvent $event): void
    {
        if (Contact::NAME !== $event->getObject()->getName()) {
            return;
        }

        // do whatever you want with $event->getUpdatedObjectMappings()
    }

    public function createContacts(InternalObjectCreateEvent $event): void
    {
        if (Contact::NAME !== $event->getObject()->getName()) {
            return;
        }
        // work whatever you want with $event->getObjectMappings()
    }
}
```

Comment (dennisameling) [reply app/bundles/IntegrationsBundle/EventListener/CompanyObjectSubscriber.php:75]: Ah okay. From https://symfony.com/doc/current/components/event_dispatcher.html#stopping-event-flow-propagation: 

> In some cases, it may make sense for a listener to prevent any other listeners from being called. In other words, the listener needs to be able to tell the dispatcher to stop all propagation of the event to future listeners (i.e. to not notify any more listeners). This can be accomplished from inside a listener via the `stopPropagation()` method.

Makes sense now 😊

**Review by mabumusa1 — APPROVED** [review]
The code is good to go after rebase

**Comment by kuzmany** [comment]
@mabumusa1 I do not see anything to block the PR and need rebase. What do you mean?

**Comment by mabumusa1** [comment]
> @mabumusa1 I do not see anything to block the PR and need rebase. What do you mean?

@kuzmany there was an issue with Phpstan, I fixed it. It does not need a rebase just a merge of the latest changes from 4.x branch

File: app/bundles/IntegrationsBundle/Sync/DAO/Mapping/UpdatedObjectMappingDAO.php
Comment (escopecz) [inline app/bundles/IntegrationsBundle/Sync/DAO/Mapping/UpdatedObjectMappingDAO.php:30]: Isn't internal object ID always int?
```suggestion
     * @var int
```

File: app/bundles/IntegrationsBundle/Sync/DAO/Mapping/UpdatedObjectMappingDAO.php
Comment (escopecz) [inline app/bundles/IntegrationsBundle/Sync/DAO/Mapping/UpdatedObjectMappingDAO.php:38]:
```suggestion
     * @param int    $internalObjectId
```

File: app/bundles/IntegrationsBundle/Sync/DAO/Mapping/UpdatedObjectMappingDAO.php
Comment (escopecz) [inline app/bundles/IntegrationsBundle/Sync/DAO/Mapping/UpdatedObjectMappingDAO.php:81]: Or can it be also null?

**Review by escopecz — COMMENTED** [review]
One question about the type.

File: app/bundles/IntegrationsBundle/EventListener/CompanyObjectSubscriber.php
Comment (escopecz) [reply app/bundles/IntegrationsBundle/EventListener/CompanyObjectSubscriber.php:75]: This should be fine. I see no other reason for stopping the propagation other than a small performance gain that it does not have to check the other object. It should not break any funciton.

**Comment by kuzmany** [comment]
Close at the moment. We will push it again once ready to M5

**Pull request closed without merging**
